src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -121,6 +131,10 @@ object Task {
     // Regular expression for matching taskIds before instance-era
     private val LegacyTaskIdRegex = """^(.+)[\._]([^_\.]+)$""".r
 
+    // Regular expression for matching resident app taskIds
+    private val ResidentTaskIdRegex = """^(.+)([\._])([^_\.]+)(\.\#)(\d+)$""".r
+    private val ResidentTaskIdAttemptSeparator = ".#"
```

Comment:
Thanks for pointing out @jdef 
Fixed in a diff on master and including the cherry-picked fix in this PR now